tenseal/tensors/__init__.py

```diff
@@ -6,39 +6,79 @@
 import _tenseal_cpp as _ts_cpp
 
 
-def bfv_vector(context, plaintext_vector):
+def bfv_vector(context, data):
     """Constructor method for the BFVVector object, which can store a list
     of integers in encrypted form, using the BFV homomorphic encryption
     scheme.
 
     Args:
         context: a TenSEALContext object, holding the encryption parameters and keys.
-        plaintext_vector: a list of integers to be encrypted.
+        data: a list of integers to be encrypted.
 
     Returns:
         BFVVector object.
     """
+    if isinstance(context, _ts_cpp.TenSEALContext) and isinstance(data, list):
+        return _ts_cpp.BFVVector(context, data)
+    """
+    Args:
+        context: a TenSEALContext object, holding the encryption parameters and keys.
+        data: serialized protobuffer.
+
+    Returns:
+        BFVVector object.
+    """
+    if isinstance(context, _ts_cpp.TenSEALContext) and isinstance(data, bytes):
+        return _ts_cpp.BFVVector(context, data)
 
-    return _ts_cpp.BFVVector(context, plaintext_vector)
+    raise "Invalid BFV input types context: {} and vector: {}".format(type(context), type(data))
```

Comment:
Sorry if I wasn't clear about this, I didn't meant we should have a single API functions, we still need both to differentiate them, but the user could have used both of them to do the same, as there was no python checking on the parameters, and Cpp overloading will be used either you pass a list or bytes, the method will return the object. However, we need different functions, `tensor` and `tensor_from` that take a list or bytes respectively, and can't take something else, the check should be done on the Python side and not in Cpp so that overloading isn't accidentally used in an undefined way.